src/CBot/CBotInstr/CBotExprLitString.cpp

```diff
@@ -134,9 +134,10 @@ CBotInstr* CBotExprLitString::Compile(CBotToken* &p, CBotCStack* pStack)
                                 }
                                 else if (maxlen == hex.length()) // unicode character
                                 {
-                                    if (val < 0xD800 || (0xDFFF < val && val < 0x110000))
+                                    plomp = CodePointToUTF8(val);
```

Comment:
Define variables as close to the place where they're needed.

You seem to worry about worse performance when creating a variable in a loop but it doesn't matter, the compiler will most certainly optimize it.